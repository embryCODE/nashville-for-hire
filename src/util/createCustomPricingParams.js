import { Money } from 'sharetribe-flex-sdk/src/types'

export default function ({ listing, bookingData }) {
  const lineItems = Object.values(bookingData).map((d) => {
    // TODO: Handle when there is no price correctly
    const unitPrice = d.price ? new Money(d.price.amount, d.price.currency) : new Money(0, 'USD')

    return {
      code: 'line-item/' + d.code,
      unitPrice,
      quantity: d.quantity,
    }
  })

  return {
    listingId: listing.id,
    lineItems,
  }
}
