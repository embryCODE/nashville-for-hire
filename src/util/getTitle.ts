import { PublicData } from '../components/NFHCustom/types'

const getTitle = (code: string, publicData: PublicData) => {
  const pricesArray = Object.values(publicData.prices)
  const realCode = code.split('/')[1]
  const price = pricesArray.find((p) => p.code === realCode)

  if (!price) throw new Error()

  return price.title
}

export default getTitle
