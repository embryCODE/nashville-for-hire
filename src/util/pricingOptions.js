export default {
  femaleVocals: {
    option0: {
      code: 'femaleVocals0',
      description: 'Lead vocal, doubled lead, multiple harmony tracks.',
      title: 'Full Vocal Package',
    },
    option1: {
      code: 'femaleVocals1',
      description: 'Lead vocal only. No harmony.',
      title: 'Lead Vocal',
    },
    option2: {
      code: 'femaleVocals2',
      description: 'Lead vocal + 1 harmony track',
      title: 'Lead Vocal + Harmony',
    },
    option3: {
      code: 'femaleVocals3',
      description: 'Backing vocals only',
      title: 'Backing Vocals',
    },
    option4: {
      code: 'femaleVocals4',
      description: 'Commercial release',
      title: 'Commercial Release',
    },
  },
  maleVocals: {
    option0: {
      code: 'maleVocals0',
      description: 'Lead vocal, doubled lead, multiple harmony tracks.',
      title: 'Full Vocal Package',
    },
    option1: {
      code: 'maleVocals1',
      description: 'Lead vocal only. No harmony.',
      title: 'Lead Vocal',
    },
    option2: {
      code: 'maleVocals2',
      description: 'Lead Vocal + 1 harmony track',
      title: 'Lead Vocal + Harmony',
    },
    option3: {
      code: 'maleVocals3',
      description: 'Backing vocals only',
      title: 'Backing Vocals',
    },
    option4: {
      code: 'maleVocals4',
      description: 'Commercial release',
      title: 'Commercial Release',
    },
  },
  songwriting: {
    option0: {
      code: 'songwriting0',
      description: 'Full cowrite via Zoom call',
      title: 'Full Cowrite',
    },
    option1: {
      code: 'songwriting1',
      description: '30 minute songwriting coaching session',
      title: '30 Minute Songwriting Coaching',
    },
    option2: {
      code: 'songwriting2',
      description: '1 hour songwriting coaching session',
      title: '1 Hour Songwriting Coaching',
    },
    option3: {
      code: 'songwriting3',
      description: 'Help finish your song (no video call)',
      title: 'Help Finish Your Song',
    },
  },
  topline: {
    option0: {
      code: 'topline0',
      description: 'Write lyrics, melody, and sing',
      title: 'Full Topline',
    },
    option1: {
      code: 'topline1',
      description: 'Sing your melody and lyrics',
      title: 'Sing Your Melody',
    },
    option2: {
      code: 'topline2',
      description: 'Write lyrics and sing your melody',
      title: 'Write Lyrics and Sing Your Melody',
    },
    option3: {
      code: 'topline3',
      description: 'Write melody and sing your lyrics',
      title: 'Write Melody and Sing Your Lyrics',
    },
  },
  acousticGuitar: {
    option0: {
      code: 'acousticGuitar0',
      description: 'Up to 4 tracks',
      title: 'Acoustic Guitar Package',
    },
    option1: {
      code: 'acousticGuitar1',
      description: 'Up to 40 seconds',
      title: 'Acoustic Guitar Solo',
    },
    option2: {
      code: 'acousticGuitar2',
      description: 'Acoustic guitar solo plus harmony line. Up to 40 seconds.',
      title: 'Acoustic Guitar Solo + Harmony',
    },
  },
  electricGuitar: {
    option0: {
      code: 'electricGuitar0',
      description: 'Up to 4 tracks (doubled)',
      title: 'Electric Guitar Package',
    },
    option1: {
      code: 'electricGuitar1',
      description: 'Up to 40 seconds',
      title: 'Electric Guitar Solo',
    },
    option2: {
      code: 'electricGuitar2',
      description: 'Electric guitar solo plus harmony line. Up to 40 seconds.',
      title: 'Electric Guitar Solo + Harmony',
    },
  },
  bass: {
    option0: {
      code: 'bass0',
      description: 'Electric bass',
      title: 'Electric Bass',
    },
    option1: {
      code: 'bass1',
      description: 'Upright bass',
      title: 'Upright Bass',
    },
    option2: {
      code: 'bass2',
      description: 'Synth bass',
      title: 'Synth Bass',
    },
  },
  drums: {
    option0: {
      code: 'drums0',
      description: 'Drum kit',
      title: 'Drum Kit',
    },
    option1: {
      code: 'drums1',
      description: 'Add percussion',
      title: 'Percussion',
    },
  },
  piano: {
    option0: {
      code: 'piano0',
      description: 'Up to 3 piano tracks on a song',
      title: 'Piano Package',
    },
    option1: {
      code: 'piano1',
      description: '1 piano track on a song',
      title: '1 Piano Track',
    },
  },
  keys: {
    option0: {
      code: 'keys0',
      description: 'Up to 3 keys tracks',
      title: 'Keys Package',
    },
    option1: {
      code: 'keys1',
      description: '1 keys track',
      title: '1 Keys Track',
    },
  },
  organ: {
    option0: {
      code: 'organ0',
      description: 'Up to 3 organ tracks',
      title: 'Organ Package',
    },
    option1: {
      code: 'organ1',
      description: '1 organ track',
      title: '1 Organ Track',
    },
  },
  synth: {
    option0: {
      code: 'synth0',
      description: 'No beat programming. Up to 4 tracks.',
      title: 'Basic Synth Arrangement',
    },
    option1: {
      code: 'synth1',
      description: 'No beat programming. Up to 8 tracks.',
      title: 'Full Synth Arrangement',
    },
    option2: {
      code: 'synth2',
      description: 'Synth bass only',
      title: 'Synth Bass',
    },
    option3: {
      code: 'synth3',
      description: '1-2 synth tracks',
      title: '1-2 Synth Tracks',
    },
  },
  beatProgramming: {
    option0: {
      code: 'beatProgramming0',
      description: 'Basic beat package',
      title: 'Basic Beat Package',
    },
    option1: {
      code: 'beatProgramming1',
      description: 'Complex beat package',
      title: 'Complex Beat Package',
    },
  },
  cello: {
    option0: {
      code: 'cello0',
      description: 'Solo cello',
      title: 'Solo Cello',
    },
    option1: {
      code: 'cello1',
      description: 'Cello trio',
      title: 'Cello Trio',
    },
    option2: {
      code: 'cello2',
      description: 'Cello quartet',
      title: 'Cello Quartet',
    },
  },
  violin: {
    option0: {
      code: 'violin0',
      description: 'Solo violin',
      title: 'Solo Violin',
    },
    option1: {
      code: 'violin1',
      description: 'Violin trio',
      title: 'Violin Trio',
    },
    option2: {
      code: 'violin2',
      description: 'Violin quartet',
      title: 'Violin Quartet',
    },
  },
  fiddle: {
    option0: {
      code: 'fiddle0',
      description: 'Solo fiddle',
      title: 'Solo Fiddle',
    },
    option1: {
      code: 'fiddle1',
      description: 'Solo viola',
      title: 'Solo Viola',
    },
    option2: {
      code: 'fiddle2',
      description: 'Viola trio',
      title: 'Viola Trio',
    },
    option3: {
      code: 'fiddle3',
      description: 'Viola quartet',
      title: 'Viola Quartet',
    },
  },
  fullStringArrangement: {
    option0: {
      code: 'fullStringArrangement0',
      description: 'Full string arrangement for trio',
      title: 'Trio',
    },
    option1: {
      code: 'fullStringArrangement1',
      description: 'Full string arrangement for quartet',
      title: 'Quartet',
    },
    option2: {
      code: 'fullStringArrangement2',
      description: 'Full string arrangement for large string ensemble',
      title: 'Large String Arrangement',
    },
  },
  pedalSteel: {
    option0: {
      code: 'pedalSteel0',
      description: 'Up to 3 pedal steel tracks on a song',
      title: 'Pedal Steel Package',
    },
    option1: {
      code: 'pedalSteel1',
      description: '1 pedal steel track on a song',
      title: '1 Pedal Steel Track',
    },
  },
  banjo: {
    option0: {
      code: 'banjo0',
      description: 'Up to 3 banjo tracks on a song',
      title: 'Banjo Package',
    },
    option1: {
      code: 'banjo1',
      description: '1 banjo track on a song',
      title: '1 Banjo Track',
    },
  },
  dobro: {
    option0: {
      code: 'dobro0',
      description: 'Up to 3 dobro tracks on a song',
      title: 'Dobro Package',
    },
    option1: {
      code: 'dobro1',
      description: '1 Dobro Track on a Song',
      title: '1 Dobro Track on a Song',
    },
  },
  lapSteel: {
    option0: {
      code: 'lapSteel0',
      description: 'Up to 3 lap steel tracks on a song',
      title: 'Lap Steel Package',
    },
    option1: {
      code: 'lapSteel1',
      description: '1 lap steel track on a song',
      title: '1 Lap Steel Track',
    },
  },
  harmonica: {
    option0: {
      code: 'harmonica0',
      description: 'Solo harmonica',
      title: 'Solo Harmonica',
    },
  },
  trumpet: {
    option0: {
      code: 'trumpet0',
      description: 'Solo trumpet',
      title: 'Solo Trumpet',
    },
    option1: {
      code: 'trumpet1',
      description: 'Trumpet section',
      title: 'Trumpet Section',
    },
  },
  trombone: {
    option0: {
      code: 'trombone0',
      description: 'Solo trombone',
      title: 'Solo Trombone',
    },
    option1: {
      code: 'trombone1',
      description: 'Trombone section',
      title: 'Trombone Section',
    },
  },
  frenchHorn: {
    option0: {
      code: 'frenchHorn0',
      description: 'Solo French Horn',
      title: 'Solo French Horn',
    },
    option1: {
      code: 'frenchHorn1',
      description: 'French Horn section',
      title: 'French Horn Section',
    },
  },
  saxophone: {
    option0: {
      code: 'saxophone0',
      description: 'Solo sax',
      title: 'Solo Sax',
    },
    option1: {
      code: 'saxophone1',
      description: 'Sax section',
      title: 'Sax Section',
    },
  },
  hornSection: {
    option0: {
      code: 'hornSection0',
      description: '3 piece horn section',
      title: '3 Piece',
    },
    option1: {
      code: 'hornSection1',
      description: '5 piece horn section',
      title: '5 Piece',
    },
  },
  mixing: {
    option0: {
      code: 'mixing0',
      description: '1-16 tracks',
      title: '1-16 Tracks',
    },
    option1: {
      code: 'mixing1',
      description: '17-48 tracks',
      title: '17-48 Tracks',
    },
    option2: {
      code: 'mixing2',
      description: '49+ tracks',
      title: '49+ Tracks',
    },
    option3: {
      code: 'mixing3',
      description: '1-2 tracks',
      title: 'Vocal Tuning 1-2 Tracks',
    },
    option4: {
      code: 'mixing4',
      description: '3-5 tracks',
      title: 'Vocal Tuning 3-5 Tracks',
    },
    option5: {
      code: 'mixing5',
      description: '6-12 tracks',
      title: 'Vocal Tuning 6-12 Tracks',
    },
  },
  mastering: {
    option0: {
      code: 'mastering0',
      description: 'Master one song',
      title: 'Master One Song',
    },
  },
  produce: {
    option0: {
      code: 'produce0',
      description: '1-3 instruments',
      title: 'Simple Production',
    },
    option1: {
      code: 'produce1',
      description: 'Full band or programmed',
      title: 'Full Production',
    },
    option2: {
      code: 'produce2',
      description: '1-2 tracks',
      title: 'Vocal Tuning 1-2 Tracks',
    },
    option3: {
      code: 'produce3',
      description: '3-5 tracks',
      title: 'Vocal Tuning 3-5 Tracks',
    },
    option4: {
      code: 'produce4',
      description: '6-12 tracks',
      title: 'Vocal Tuning 6-12 Tracks',
    },
    option5: {
      code: 'produce5',
      description: 'Add mixing to simple production',
      title: 'Simple mix',
    },
    option6: {
      code: 'produce6',
      description: 'Add mixing to full production',
      title: 'Full mix',
    },
  },
}
