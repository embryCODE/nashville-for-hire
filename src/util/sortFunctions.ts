import { Price } from '../components/NFHCustom/types'

export const sortByPriceCode = (a: Price, b: Price) => {
  if (a.code.includes('custom')) return 1
  if (b.code.includes('custom')) return -1
  return 0
}

export const sortByPricesAndQuantityCode = ([, a]: [string, Price], [, b]: [string, Price]) => {
  if (a.code.includes('custom')) return 1
  if (b.code.includes('custom')) return -1
  return 0
}
