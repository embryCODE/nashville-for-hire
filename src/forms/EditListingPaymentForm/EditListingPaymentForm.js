import React from 'react'
import { bool, func, shape, string } from 'prop-types'
import { compose } from 'redux'
import { Form as FinalForm } from 'react-final-form'
import { intlShape, injectIntl, FormattedMessage } from '../../util/reactIntl'
import classNames from 'classnames'
import config from '../../config'
import { LINE_ITEM_NIGHT, LINE_ITEM_DAY, propTypes } from '../../util/types'
import * as validators from '../../util/validators'
import { formatMoney } from '../../util/currency'
import { types as sdkTypes } from '../../util/sdkLoader'
import { Button, Form, FieldCurrencyInput, FieldTextInput } from '../../components'
import css from './EditListingPaymentForm.css'
import { composeValidators, required } from '../../util/validators'

const { Money } = sdkTypes

export const EditListingPaymentFormComponent = (props) => {
  return (< FinalForm
      {...
        props
      }
      render={(formRenderProps) => {
        const {
          serviceType,
          className,
          disabled,
          ready,
          handleSubmit,
          intl,
          invalid,
          pristine,
          saveActionMsg,
          updated,
          updateInProgress,
          fetchErrors,
        } = formRenderProps

        const unitType = config.bookingUnitType
        const isNightly = unitType === LINE_ITEM_NIGHT
        const isDaily = unitType === LINE_ITEM_DAY

        const translationKey = isNightly
          ? 'EditListingPaymentForm.pricePerNight'
          : isDaily
            ? 'EditListingPaymentForm.pricePerDay'
            : 'EditListingPaymentForm.pricePerUnit'

        const pricePerUnitMessage = intl.formatMessage({
          id: translationKey,
        })

        const pricePlaceholderMessage = intl.formatMessage({
          id: 'EditListingPaymentForm.priceInputPlaceholder',
        })

        const priceRequired = validators.required(
          intl.formatMessage({
            id: 'EditListingPaymentForm.priceRequired',
          }),
        )
        const minPrice = new Money(config.listingMinimumPriceSubUnits, config.currency)
        const minPriceRequired = validators.moneySubUnitAmountAtLeast(
          intl.formatMessage(
            {
              id: 'EditListingPaymentForm.priceTooLow',
            },
            {
              minPrice: formatMoney(intl, minPrice),
            },
          ),
          config.listingMinimumPriceSubUnits,
        )
        const priceValidators = config.listingMinimumPriceSubUnits
          ? validators.composeValidators(priceRequired, minPriceRequired)
          : priceRequired

        const classes = classNames(css.root, className)
        const submitReady = (updated && pristine) || ready
        const submitInProgress = updateInProgress
        const submitDisabled = invalid || disabled || submitInProgress
        const { updateListingError, showListingsError } = fetchErrors || {}

        const questionATitle = intl.formatMessage({ id: 'EditListingPaymentForm.questionATitle' })
        const questionATitlePlaceholderMessage = intl.formatMessage({
          id: 'EditListingPaymentForm.questionATitlePlaceholder',
        })
        const questionBTitle = intl.formatMessage({ id: 'EditListingPaymentForm.questionBTitle' })
        const questionBTitlePlaceholderMessage = intl.formatMessage({
          id: 'EditListingPaymentForm.questionBTitlePlaceholder',
        })

        return (
          <Form onSubmit={handleSubmit} className={classes}>
            {updateListingError ? (
              <p className={css.error}>
                <FormattedMessage id="EditListingPaymentForm.updateFailed"/>
              </p>
            ) : null}
            {showListingsError ? (
              <p className={css.error}>
                <FormattedMessage id="EditListingPaymentForm.showListingFailed"/>
              </p>
            ) : null}
            <FieldTextInput
              id="question_a"
              name="question_a"
              className={css.title}
              type="text"
              label={questionATitle}
              placeholder={questionATitlePlaceholderMessage}
              // validate={composeValidators(required(titleRequiredMessage), maxLength60Message)}
              autoFocus
            />

            <FieldTextInput
              id="question_b"
              name="question_b"
              className={css.title}
              type="text"
              label={questionBTitle}
              placeholder={questionBTitlePlaceholderMessage}
              // validate={composeValidators(required(titleRequiredMessage), maxLength60Message)}
              autoFocus
            />
            <Button
              className={css.submitButton}
              type="submit"
              inProgress={submitInProgress}
              disabled={submitDisabled}
              ready={submitReady}
            >
              {saveActionMsg}
            </Button>
          </Form>
        )
      }
      }
    />
  )
}
EditListingPaymentFormComponent.defaultProps = { fetchErrors: null }

EditListingPaymentFormComponent.propTypes = {
  serviceType: string.isRequired,
  intl: intlShape.isRequired,
  onSubmit: func.isRequired,
  saveActionMsg: string.isRequired,
  disabled: bool.isRequired,
  ready: bool.isRequired,
  updated: bool.isRequired,
  updateInProgress: bool.isRequired,
  fetchErrors: shape({
    showListingsError: propTypes.error,
    updateListingError: propTypes.error,
  }),
}

export default compose(injectIntl)(EditListingPaymentFormComponent)
