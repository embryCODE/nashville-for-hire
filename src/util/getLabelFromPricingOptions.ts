import pricingOptions from './pricingOptions'

const unnestedPricingOptions = Object.values(pricingOptions)
  .flat()
  .map((po) => Object.values(po).flat())
  .flat()

const getLabelFromPricingOptions = (code: string): string => {
  const pricingOption = unnestedPricingOptions.find((po) => code.includes(po.code))
  return pricingOption ? pricingOption.label : ''
}

export default getLabelFromPricingOptions
