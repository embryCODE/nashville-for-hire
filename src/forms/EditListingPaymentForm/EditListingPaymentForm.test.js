import React from 'react'
import { renderDeep } from '../../util/test-helpers'
import { fakeIntl } from '../../util/test-data'
import EditListingPaymentForm from './EditListingPaymentForm'

const noop = () => null

describe('EditListingPaymentForm', () => {
  it('matches snapshot', () => {
    const tree = renderDeep(
      <EditListingPaymentForm
        intl={fakeIntl}
        dispatch={noop}
        onSubmit={(v) => v}
        saveActionMsg="Save price"
        updated={false}
        updateInProgress={false}
        disabled={false}
        ready={false}
      />,
    )
    expect(tree).toMatchSnapshot()
  })
})
