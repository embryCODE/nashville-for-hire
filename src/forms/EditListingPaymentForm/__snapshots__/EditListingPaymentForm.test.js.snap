// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditListingPaymentForm matches snapshot 1`] = `
<form
  action="/"
  className=""
  method="post"
  onSubmit={[Function]}
>
  <div
    className=""
  >
    <label
      htmlFor="payPalEmail"
    >
      EditListingPaymentForm.questionATitle
    </label>
    <input
      autoFocus={true}
      className=""
      id="payPalEmail"
      name="payPalEmail"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="EditListingPaymentForm.questionATitlePlaceholder"
      type="text"
      value=""
    />
  </div>
  <div
    className=""
  >
    <label
      htmlFor="phoneNumber"
    >
      EditListingPaymentForm.questionBTitle
    </label>
    <input
      autoFocus={true}
      className=""
      id="phoneNumber"
      name="phoneNumber"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="EditListingPaymentForm.questionBTitlePlaceholder"
      type="text"
      value=""
    />
  </div>
  <button
    className=""
    disabled={false}
    type="submit"
  >
    Save price
  </button>
</form>
`;
